### PR TITLE
Use `PackedVector4Array` instead of float array for vec4 array uniform

### PR DESCRIPTION
**Pull request by Chaosus**
I think it's a good fix.

### CODE REVIEW

**Comment by akien-mga** [comment]
I had the same change suggested in https://github.com/godotengine/godot/pull/85474#issuecomment-2091883476 indeed, but wasn't sure it was the only place where it needs to be changed.

If you've tested that the functionality still works as expected, including loading vec4 array uniforms saved in previous engine versions, this should be good to go.

**Comment by Chaosus** [comment]
I think there is a minor compatibility breakage, not sure if this is viable for 4.3. Should I wrap it with `DISABLE_DEPRECATED`?

File: scene/resources/material.cpp [hunk 1]
```diff
@@ -324,6 +324,28 @@ void ShaderMaterial::_get_property_list(List<PropertyInfo> *p_list) const {
...
+					for (int i = 0; i < array.size(); i += 4) {
+						Vector4 v = Vector4();
+
+						v.x = array[i];
+						v.y = (i + 1 < array.size()) ? array[i + 1] : 0.0f;
+						v.z = (i + 2 < array.size()) ? array[i + 2] : 0.0f;
+						v.w = (i + 3 < array.size()) ? array[i + 3] : 0.0f;
+
+						varray.push_back(v);
+					}
```
Comment (akien-mga) [inline scene/resources/material.cpp:342]: Can maybe be simplified like this, so we skip the last iteration if it's truncated. It shouldn't happen anyway as that's a degenerate case, so it's fine IMO to cut corners in this case. (Didn't test whether my `i + 3 < array.size()` bound logic works.)
```suggestion
					for (int i = 0; i + 3 < array.size(); i += 4) {
						varray.push_back(Vector4(array[i], array[i + 1], array[i + 2], array[i + 3]);
					}
```

**Review by akien-mga — APPROVED** [review]
Looks good to me with added compat code.

File: scene/resources/material.cpp (hunk 1 above)
Comment (Chaosus) [reply scene/resources/material.cpp]: Yes, it works.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**